drivers/pinctrl/pinctrl_nrf.c

```diff
@@ -367,6 +370,24 @@ int pinctrl_configure_pins(const pinctrl_soc_pin_t *pins, uint8_t pin_cnt,
 #if NRF_GPIO_HAS_CLOCKPIN
 			nrf_gpio_pin_clock_set(pin, NRF_GET_CLOCKPIN_ENABLE(pins[i]));
 #endif
+#ifdef CONFIG_SOC_NRF54H20_GPD
+			/* if a pin (peripheral) belongs to FAST_ACTIVE1, keep SLOW_ACTIVE alive */
+			if (NRF_GET_GPD_FAST_ACTIVE1(pins[i]) == 1U) {
+				int ret;
+
+				if (NRF_GET_LP(pins[i]) == NRF_LP_ENABLE) {
+					ret = nrf_gpd_release(NRF_GPD_SLOW_ACTIVE);
+					if (ret < 0) {
+						return ret;
+					}
+				} else {
+					ret = nrf_gpd_request(NRF_GPD_SLOW_ACTIVE);
```

Comment:
Now I'm thinking if we should do the GPIO.RETAINSET trick here instead. If we keep GDP requested, it would prevent entering low-power modes

Reply:
Perhaps not instead, but alongside. We should check if the pin is going to be controlled by the peripheral. If so, request GPD. If not, set GPIO.RETAIN.
And then we would need to identify where to clear GPIO.RETAIN.